fastestimator/op/tensorop/gather.py

```diff
@@ -57,9 +57,13 @@ def forward(self, data: List[Tensor], state: Dict[str, Any]) -> List[Tensor]:
         results = []
         for idx, tensor in enumerate(inputs):
             # Check len(indices[0]) since an empty indices element is used to trigger the else
-            if len(indices) > idx and len(indices[0]) > 0:
+            if isinstance(indices[0], tf.Tensor) or isinstance(indices[0], torch.Tensor):
+                elem_len = indices[0].shape[0]
```

Comment:
interesting. what was the issue with calling len(indices[0]) on a Tensor?

Reply:
It gives an error on multi-gpu system. Works fine on single gpu.
TypeError: len is not well defined for symbolic Tensors. (transpose:0) Please call `x.shape` rather than `len(x)` for shape information

Reply:
huh. cool